### PR TITLE
collect uid dependency for incremental.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 1]
```diff
@@ -208,6 +210,39 @@ private void BuildCore(DocumentBuildParameters parameters)
...
+            {
+                foreach (var m in hostService.Models)
+                {
+                    if (m.Type == DocumentType.Overwrite)
+                    {
+                        continue;
+                    }
+                    context.DependencyGraph.ReportDependency(
+                        m.OriginalFileAndType.File,
+                        GetFilesFromUids(context, m.LinkToUids));
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: toList()?

File: src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs [hunk 2]
```diff
@@ -81,8 +81,8 @@ select new
             {
                 var vm = pair.item;
                 Merge(vm, ovm, ovms[0].fileModel);
-                ((HashSet<string>)pair.model.Properties.LinkToUids).UnionWith((HashSet<string>)od[0].Properties.LinkToUids);
-                ((HashSet<string>)pair.model.Properties.LinkToFiles).UnionWith((HashSet<string>)od[0].Properties.LinkToFiles);
+                pair.model.LinkToUids = pair.model.LinkToUids.Union(od[0].LinkToUids);
+                pair.model.LinkToFiles = pair.model.LinkToFiles.AddRange(od[0].LinkToFiles);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs]: why allow duplicate linktofile?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: why, only one consumer.

File: src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs]: fixed.

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**